### PR TITLE
Code coverage

### PR DESCRIPTION
**Pull request by nmarkovic04**

### CODE REVIEW

File: AutocompleteClient/FW/UI/DefaultSearchItemCell.xib [hunk 1]
```diff
@@ -35,7 +35,7 @@
             <connections>
                 <outlet property="labelText" destination="swl-kL-qeP" id="B4N-VJ-CRQ"/>
             </connections>
-            <point key="canvasLocation" x="-49" y="104"/>
+            <point key="canvasLocation" x="-49.600000000000001" y="103.44827586206897"/>
```
Comment (rubikzube) [inline AutocompleteClient/FW/UI/DefaultSearchItemCell.xib]: should these x/y coordinates be this specific?

File: AutocompleteClient.xcworkspace/xcshareddata/WorkspaceSettings.xcsettings [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+	<key>BuildSystemType</key>
+	<string>Original</string>
```
Comment (rubikzube) [inline AutocompleteClient.xcworkspace/xcshareddata/WorkspaceSettings.xcsettings]: What do we gain from the `original` build system type?

File: AutocompleteClient/FW/UI/ViewModel/AutocompleteViewModel.swift [hunk 3]
```diff
@@ -8,7 +8,22 @@
 
 import UIKit
 
-public class AutocompleteViewModel {
+public protocol AbstractAutocompleteViewModel{
```
Comment (rubikzube) [inline AutocompleteClient/FW/UI/ViewModel/AutocompleteViewModel.swift]: should this protocol be in a separate file?

File: AutocompleteClient/Utils/String/String+SearchSuggestionsSection.swift [hunk 4]
```diff
@@ -8,7 +8,7 @@
 
 import Foundation
 
-extension String {
+public extension String {
```
Comment (rubikzube) [inline AutocompleteClient/Utils/String/String+SearchSuggestionsSection.swift]: If we publicly extend string would that extension propagate to consumers?

File: AutocompleteClientTests/HelperTests/SearchSuggestionStringConversionTests.swift [hunk 5]
```diff
@@ -0,0 +1,22 @@
+//
+//  SearchSuggestionStringConversionTests.swift
+//  AutocompleteClientTests
+//
+//  Created by Nikola Markovic on 1/4/19.
+//  Copyright © 2019 xd. All rights reserved.
```
Comment (rubikzube) [inline AutocompleteClientTests/HelperTests/SearchSuggestionStringConversionTests.swift]: Need to revise comments

File: AutocompleteClientTests/HelperTests/SearchBarTextFieldTests.swift [hunk 6]
```diff
@@ -0,0 +1,29 @@
+//
+//  SearchBarTextFieldTests.swift
+//  AutocompleteClientTests
+//
+//  Created by Nikola Markovic on 1/4/19.
+//  Copyright © 2019 xd. All rights reserved.
```
Comment (rubikzube) [inline AutocompleteClientTests/HelperTests/SearchBarTextFieldTests.swift]: Need to revise comments

File: AutocompleteClientTests/HelperTests/UIViewAnimationTests.swift [hunk 7]
```diff
@@ -0,0 +1,42 @@
+//
+//  UIViewAnimationTests.swift
+//  AutocompleteClientTests
+//
+//  Created by Nikola Markovic on 1/4/19.
+//  Copyright © 2019 xd. All rights reserved.
```
Comment (rubikzube) [inline AutocompleteClientTests/HelperTests/UIViewAnimationTests.swift]: Need to revise comments

File: AutocompleteClient/FW/UI/DefaultSearchItemCell.xib (hunk 1 above)
Comment (nmarkovic04) [reply AutocompleteClient/FW/UI/DefaultSearchItemCell.xib]: It's just the position of the top view in the canvas. It's pretty irrelevant.

File: AutocompleteClient.xcworkspace/xcshareddata/WorkspaceSettings.xcsettings (hunk 2 above)
Comment (nmarkovic04) [reply AutocompleteClient.xcworkspace/xcshareddata/WorkspaceSettings.xcsettings]: Currently, this is the most safest way to build the project.

File: AutocompleteClient/Utils/String/String+SearchSuggestionsSection.swift (hunk 4 above)
Comment (nmarkovic04) [reply AutocompleteClient/Utils/String/String+SearchSuggestionsSection.swift]: Yes.

File: AutocompleteClientTests/HelperTests/SearchSuggestionStringConversionTests.swift (hunk 5 above)
Comment (nmarkovic04) [reply AutocompleteClientTests/HelperTests/SearchSuggestionStringConversionTests.swift]: Done.

File: AutocompleteClient/FW/UI/ViewModel/AutocompleteViewModel.swift (hunk 3 above)
Comment (nmarkovic04) [reply AutocompleteClient/FW/UI/ViewModel/AutocompleteViewModel.swift]: Done.

**Comment by nmarkovic04** [comment]
@rubikzube I moved all the items from HelperTests to appropriate folders. The only folder that doesn't match the framework project structure is TestUtilsTests which tests classes inside Test Utils - classes that are only contained in the test target.

File: AutocompleteClient/Utils/String/String+SearchSuggestionsSection.swift (hunk 4 above)
Comment (rubikzube) [reply AutocompleteClient/Utils/String/String+SearchSuggestionsSection.swift]: Is there another way to do this?  I'm not sure that consuming apps would love that we're extending common classes like string

**Pull request merged**